### PR TITLE
More gradle cleanups

### PR DESCRIPTION
**Pull request by bogdandrutu**
* Only projects with the plugin maven-publish will be published
* Only projects with the plugin me.champeau.gradle.jmh will build/run benchmarcs.
* Enforce jmhreport always after jmh even if project does not include jmhreport plugin (it is automatically included).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -151,27 +162,14 @@ subprojects {
 
         // The ErrorProne plugin defaults to the latest, which would break our
         // build if error prone releases a new version with a new check
-        errorprone "com.google.errorprone:error_prone_core:${errorProneVersion}"
-        errorproneJavac "com.google.errorprone:javac:${errorProneJavacVersion}"
-
-        if (JavaVersion.current().isJava9() || JavaVersion.current().isJava10() || JavaVersion.current().isJava11()) {
-            // Workaround for @javax.annotation.Generated
-            // see: https://github.com/grpc/grpc-java/issues/3633
-            compile("javax.annotation:javax.annotation-api:1.3.2")
```
Comment (jkwatson) [inline build.gradle]: I think the PR build is broken because this bit needs to be here....

Comment (bogdandrutu) [reply build.gradle]: Fixed, thanks :)

File: build.gradle [hunk 2]
```diff
@@ -108,11 +100,14 @@ subprojects {
                 auto_value_annotation: "com.google.auto.value:auto-value-annotations:${autoValueVersion}",
                 disruptor            : 'com.lmax:disruptor:3.4.2',
```
Comment (thisthat) [inline build.gradle]: Since everything has a variable for its version, should we add one here as well?

File: build.gradle [hunk 3]
```diff
@@ -108,11 +100,14 @@ subprojects {
...
                 errorprone_annotation: "com.google.errorprone:error_prone_annotations:${errorProneVersion}",
+                errorprone_core      : "com.google.errorprone:error_prone_core:${errorProneVersion}",
+                errorprone_javac     : "com.google.errorprone:javac:${errorProneJavacVersion}",
                 grpc_api             : "io.grpc:grpc-api:${grpcVersion}",
                 grpc_context         : "io.grpc:grpc-context:${grpcVersion}",
                 grpc_protobuf        : "io.grpc:grpc-protobuf:${grpcVersion}",
                 grpc_stub            : "io.grpc:grpc-stub:${grpcVersion}",
                 guava                : "com.google.guava:guava:${guavaVersion}",
+                javax_annotations    : "javax.annotation:javax.annotation-api:1.3.2",
```
Comment (thisthat) [inline build.gradle]: Same comment here

File: exporters/otprotocol/build.gradle [hunk 4]
```diff
@@ -10,8 +10,9 @@ ext.moduleName = "io.opentelemetry.exporters.otprotocol"
 
 dependencies {
     api project(':opentelemetry-proto'),
-            project(':opentelemetry-sdk'),
-            libraries.grpc_protobuf,
+            project(':opentelemetry-sdk')
+
+    implementation libraries.grpc_protobuf,
```
Comment (thisthat) [inline exporters/otprotocol/build.gradle]: good catch :wink:

File: build.gradle (hunk 2 above)
Comment (bogdandrutu) [reply build.gradle]: I was focused on making sure that all dependencies are in the libraries to have a one place where we can update all deps. I think we can do that in the next PR.

File: build.gradle (hunk 3 above)
Comment (bogdandrutu) [reply build.gradle]: Proposing to do that in the next PR.

**Pull request merged**